Dockerfile-golang

```diff
@@ -2,4 +2,4 @@ FROM golang:1.16-alpine
 
 # Only install hard dependencies
 RUN apk add --no-cache gcc git musl-dev make && \
-    go get -v -u gotest.tools/gotestsum
```

Comment:
Maybe can leave the `-v -u` flags?

"The -u flag instructs get to use the network to update the named packages
and their dependencies. By default, get uses the network to check out
missing packages but does not use it to look for updates to existing packages." ([source](https://pkg.go.dev/cmd/go/internal/get))

Reply:
I removed this because it was stopping codefresh from building the build image as seen below: 

```
Removing intermediate container 83e96fae1b01                                                                                                         
The command '/bin/sh -c apk add --no-cache gcc git musl-dev make &&     go get -v -u gotest.tools/gotestsum' returned a non-zero code: 2             
[SYSTEM]                                                                                                                                             
 Message             Failed to build image: dsaidgovsg/registrywatcher-builder:latest                                                                
 Caused by           Container for step title: Build builder image, step type: build, operation: Building image. Failed                              
                     with exit code: 2                                                                                                               
 Documentation Link  https://codefresh.io/docs/docs/codefresh-yaml/steps                                                                             
 Exit code           2                                                                                                                               
 Name                NonZeroExitCodeError        
 ```
https://g.codefresh.io/build/6425178fff4721215410765a?step=build_builder&tab=output&logs=terminal

Reply:
could be a golang version issue
```note: module requires Go 1.17 ```

Reply:
[master failing](https://g.codefresh.io/build/6426a4599398eddc82a80c39), nothing to do with the changes in this PR. need to do some maintenance on the repo